src/ng/cacheFactory.js

```diff
@@ -223,6 +223,8 @@ function $CacheFactoryProvider() {
             delete lruHash[key];
           }
 
+          if (!(key in data)) return;
```

Comment:
Change https://github.com/angular/angular.js/blob/master/src/ng/cacheFactory.js#L96 and https://github.com/angular/angular.js/blob/master/src/ng/cacheFactory.js#L240 to `data = createMap();`